openslides_backend/action/actions/option/update.py

```diff
@@ -191,3 +195,21 @@ def check_state_change(
         if instance.get("Y") or instance.get("N") or instance.get("A"):
             return True
         return False
+
+    def check_permissions(self, instance: Dict[str, Any]) -> None:
+        _, poll, _ = self._get_poll(instance["id"])
+        content_object_id = poll.get("content_object_id", "")
+        meeting_id = poll["meeting_id"]
+
+        if content_object_id.startswith("motion" + KEYSEPARATOR):
+            self._check_perm(Permissions.Motion.CAN_MANAGE_POLLS, meeting_id)
+        elif content_object_id.startswith("assignment" + KEYSEPARATOR):
+            self._check_perm(Permissions.Assignment.CAN_MANAGE, meeting_id)
+        else:
+            self._check_perm(Permissions.Poll.CAN_MANAGE, meeting_id)
+
+    def _check_perm(self, perm: Permission, meeting_id: int) -> None:
+        if not has_perm(self.datastore, self.user_id, perm, meeting_id):
+            msg = f"You are not allowed to perform action {self.name}."
+            msg += f" Missing permission: {perm}"
+            raise PermissionDenied(msg)
```

Comment:
Please deduplicate this code with the poll permission mixin. Either extract this into its own seperate function which is then called in the mixin, split the mixin into two (one which holds the actual code and one which overrides `check_permissions`), etc.

Reply:
Reduced the dublication.
I added check_poll_or_option_permission and change _check_perm into an function.
Use these functions to reduce the code dublication.